cmd/server/pactasrv/initiative_invitation.go

```diff
@@ -0,0 +1,118 @@
+// (DELETE /initiative-invitation/{id})
+func (s *Server) DeleteInitiativeInvitation(ctx context.Context, request api.DeleteInitiativeInvitationRequestObject) (api.DeleteInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	err := s.DB.DeleteInitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to delete initiative invitation", zap.Error(err))
+	}
+	return api.DeleteInitiativeInvitation204Response{}, nil
+}
+
+// Returns the initiative invitation from this id, if it exists
+// (GET /initiative-invitation/{id})
+func (s *Server) GetInitiativeInvitation(ctx context.Context, request api.GetInitiativeInvitationRequestObject) (api.GetInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	ii, err := s.DB.InitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve initiative invitation", zap.Error(err))
+	}
+	result, err := conv.InitiativeInvitationToOAPI(ii)
+	if err != nil {
+		return nil, err
+	}
+	return api.GetInitiativeInvitation200JSONResponse(*result), nil
+}
+
+// Claims this initiative invitation, if it exists
+// (POST /initiative-invitation/{id})
+func (s *Server) ClaimInitiativeInvitation(ctx context.Context, request api.ClaimInitiativeInvitationRequestObject) (api.ClaimInitiativeInvitationResponseObject, error) {
+	userID, err := getUserID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = s.DB.Transactional(ctx, func(tx db.Tx) error {
+		ii, err := s.DB.InitiativeInvitation(tx, pacta.InitiativeInvitationID(request.Id))
+		if err != nil {
+			return fmt.Errorf("looking up initiative invite: %w", err)
+		}
+		if !ii.UsedAt.IsZero() || ii.UsedBy != nil {
+			if ii.UsedBy != nil && ii.UsedBy.ID == userID {
+				// We don't return an error if the same user tries to claim the same invitation twice,
+				// which might happen by accident, but wouldn't impact the state of the initiative memberships.
+				// We may want to log this, though.
+				return nil
+			} else {
+				return fmt.Errorf("initiative is already used: %+v", ii)
+			}
+		}
+		err = s.DB.UpdateInitiativeInvitation(tx, ii.ID,
+			db.SetInitiativeInvitationUsedAt(time.Now()),
+			db.SetInitiativeInvitationUsedBy(userID))
+		if err != nil {
+			return fmt.Errorf("updating initiative invite: %w", err)
+		}
+		err = s.DB.UpdateInitiativeUserRelationship(tx, ii.Initiative.ID, userID,
+			db.SetInitiativeUserRelationshipMember(true))
+		if err != nil {
+			return fmt.Errorf("creating initiative membership: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, oapierr.Internal("failed to claim initiative invitation", zap.Error(err))
+	}
+	return api.ClaimInitiativeInvitation204Response{}, nil
+}
+
+// Returns all initiative invitations associated with the initiative
+// (GET /initiative/{id}/invitations)
```

Comment:
Why is this `GET /initiative/{id}/invitations` and not `GET /initiative-invitation`? I think the standard RESTful approach would be the latter, [ex](https://cloud.google.com/apis/design/standard_methods)

Reply:
We won't want that for public access. I think keeping them distinct will make authz much easier.